rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -202,6 +220,34 @@ private CachingRlsLbClient(Builder builder) {
             lbPolicyConfig.getLoadBalancingPolicy(), childLbResolvedAddressFactory,
             childLbHelperProvider,
             new BackoffRefreshListener());
+
+    cacheEntriesGaugeRegistration = helper.getMetricRecorder()
+        .registerBatchCallback(new BatchCallback() {
+          @Override
+          public void accept(BatchRecorder recorder) {
+            int estimatedSize;
+            synchronized (lock) {
+              estimatedSize = linkedHashLruCache.estimatedSize();
+            }
+            recorder.recordLongGauge(CACHE_ENTRIES_GAUGE, estimatedSize,
+                Arrays.asList(helper.getChannelTarget(), rlsConfig.lookupService(),
+                    metricsInstanceUuid), Collections.emptyList());
+          }
+        }, CACHE_ENTRIES_GAUGE);
+    cacheSizeGaugeRegistration = helper.getMetricRecorder()
```

Comment:
This is a _batch_ callback to allow you to grab the lock once and grab multiple pieces of data. We'll want to combine these together, but it is okay for now.

Reply:
All batched up now.